calico_containers/calicoctl.py

```diff
@@ -1648,275 +84,152 @@ def permission_denied_error(conn_error):
-                else:
-                    ip_pool_show(ip_version)
-        elif arguments["bgppeer"]:
-            if arguments["add"]:
-                bgppeer_add(arguments["<PEER_IP>"], ip_version,
-                            arguments["<AS_NUM>"])
-            elif arguments["remove"]:
-                bgppeer_remove(arguments["<PEER_IP>"], ip_version)
-            elif arguments["show"]:
-                if not ip_version:
-                    bgppeer_show("v4")
-                    bgppeer_show("v6")
-                else:
-                    bgppeer_show(ip_version)
-        elif arguments["container"]:
-            if arguments["endpoint-id"]:
-                container_endpoint_id_show(arguments["<CONTAINER>"])
-            elif arguments["ip"]:
-                if arguments["add"]:
-                    container_ip_add(arguments["<CONTAINER>"],
-                                     arguments["<IP>"],
-                                     ip_version,
-                                     arguments["--interface"])
-                elif arguments["remove"]:
-                    container_ip_remove(arguments["<CONTAINER>"],
-                                        arguments["<IP>"],
-                                        ip_version,
-                                        arguments["--interface"])
-                else:
-                    if arguments["add"]:
-                        container_add(arguments["<CONTAINER>"],
-                                      arguments["<IP>"],
-                                      arguments["--interface"])
-                    if arguments["remove"]:
-                        container_remove(arguments["<CONTAINER>"])
-            else:
-                if arguments["add"]:
-                    container_add(arguments["<CONTAINER>"],
-                                  arguments["<IP>"],
-                                  arguments["--interface"])
-                if arguments["remove"]:
-                    container_remove(arguments["<CONTAINER>"])
-        elif arguments["bgp-node-mesh"]:
-            if arguments["on"] or arguments["off"]:
-                set_bgp_node_mesh(arguments["on"])
-            else:
-                show_bgp_node_mesh()
-        elif arguments["default-node-as"]:
-            if arguments["<AS_NUM>"]:
-                set_default_node_as(arguments["<AS_NUM>"])
-            else:
-                show_default_node_as()
+    # TODO: come up with a better way to default to --help when no opts provided
+    if len(sys.argv) == 1:
+        sys.argv.append('--help')
+
+    # Run command through initial docopt processing to determine subcommand
+    command_args = docopt(__doc__, options_first=True)
+
+    # Group the additional args together and forward them along
+    argv = [command_args['<command>']] + command_args['<args>']
+
+    # Dispatch the appropriate subcommand
+    try:
+        command = command_args['<command>']
+        if command == 'node':
```

Comment:
This if block is clunky, but the only other option I see is something like this:
```
elif args['<command>'] in 'container endpoint pool node'.split():
        exit(call(['python', 'calico_ctl/%s.py' % args['<command>']] + 
```
This just screams "vulernability" to me...